### PR TITLE
Add billingEnabled check to disabled bools

### PR DESCRIPTION
**Pull request by petedejoy**
I noticed while playing around this morning that a lot of my `disabled` components didn't have a billingEnabled dependency. This caused parts of the intended Cloud trial UX to apply to "enterprise mode" (ie. `STRIPE__ENABLED="false"` in a houston .env). This PR adds a check on the workspace's `billingEnabled` field to some of the ternaries that decide whether or not we should disable a UI component, preventing EE customers from running into UI issues.

### CODE REVIEW

File: src/modules/deployments/DeploymentConfigure/CreateForm.js [hunk 1]
```diff
@@ -23,9 +23,11 @@ class CreateForm extends React.Component {
 
   renderConfig() {
     const { form, deploymentConfig, deployments } = this.props
-    const disabled = deployments[0]
-      ? isTrialing(deployments[0].workspace)
-      : null
+    const disabled =
+      deployments[0] && deployments[0].workspace.billingEnabled
```
Comment (petedejoy) [inline src/modules/deployments/DeploymentConfigure/CreateForm.js]: I know this is verbose and not ideal or consistent with some other components in this module, but it was truly the best way to get the data we need here. Hoping Sam's refactor will clean all of this up.

**Comment by schnie** [comment]
@petedejoy should we consolidate this chunk of code to a shared function somewhere?

**Comment by petedejoy** [comment]
That was the idea behind the `isTrialing` function originally, but the logic has developed a bit further than I anticipated. Let me try to extend some of that functionality to clean all this up.

**Comment by petedejoy** [comment]
Wow that was...easier 😅

**Pull request merged**